src/recipe_dictionary.cpp

```diff
@@ -132,13 +132,9 @@ std::vector<const recipe *> recipe_subset::recent() const
 
     for( auto rec_id = uistate.recent_recipes.rbegin(); rec_id != uistate.recent_recipes.rend();
          ++rec_id ) {
-        std::find_if( recipes.begin(), recipes.end(), [&rec_id, &res]( const recipe * r ) {
-            if( !*r || *rec_id != r->ident() || r->obsolete ) {
-                return false;
-            }
-
-            res.push_back( r );
-            return true;
+        std::copy_if( recipes.begin(), recipes.end(), std::back_inserter( res ),
```

Comment:
Nice and idiomatic :+1:

Reply:
Yes, a similar code turned out to be a few lines below.